packages/core/src/js/sdk.tsx

```diff
@@ -62,6 +63,40 @@ export function init(passedOptions: ReactNativeOptions): void {
     enableSyncToNative(getIsolationScope());
   }
 
+  const getURLFromDSN = (dsn: string | null): string | undefined => {
+    if (!dsn) {
+      return undefined;
+    }
+    const url = new URL(dsn);
+    return `${url.protocol}//${url.host}`;
+  };
+
+  const userBeforeBreadcrumb = safeFactory(passedOptions.beforeBreadcrumb, { loggerMessage: 'The beforeBreadcrumb threw an error' });
+
+  // Exclude Dev Server and Sentry Dsn request from Breadcrumbs
+  const devServerUrl = getDevServer()?.url;
+  const dsn = getURLFromDSN(passedOptions.dsn);
+  const defaultBeforeBreadcrumb = (breadcrumb: Breadcrumb, _hint?: BreadcrumbHint): Breadcrumb | null => {
+    const type = breadcrumb.type || '';
+    const url = typeof breadcrumb.data?.url === 'string' ? breadcrumb.data.url : '';
+    if (type === 'http' && ((devServerUrl && url.includes(devServerUrl)) || (dsn && url.includes(dsn)))) {
```

Comment:
i: We could use startsWith for url and dsn, what do you think? includes will spend more time checking the entire string to see if any substring matches it where startsWith only checks the start of the string only.



```suggestion
    if (type === 'http' && ((devServerUrl && url.startsWith(devServerUrl)) || (dsn && url.startsWith(dsn)))) {
```

example:
` ` `
const dsn = "https://sentry.io";
url = "blablahttps://sentry.io";
console.log(url.includes(dsn));
// true
console.log(url.startsWith(dsn));
// false
` ` `

Reply:
That sounds like a good idea 👍

Reply:
Also updated the native side with c05c3c2679400cf285d5d3869df1eab5e3e61b9d.
Thank you for the suggestion @lucas-zimerman 🙇